servers/muxsrv/server.go

```diff
@@ -0,0 +1,42 @@
+// Package muxsrv implements a multiplexed TCP server, which multiplexes the GD2 rest, grpc and sunrpc servers
+package muxsrv
+
+import (
+	"net"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/soheilhy/cmux"
+	config "github.com/spf13/viper"
+)
+
+// MuxSrv implements the suture.Sever for the GD2 multiplexed server
+type muxSrv struct {
+	l net.Listener
+	m cmux.CMux
+}
+
+// newMuxSrv returns a multiplexed server with the multiplexed listeners already setup
+func newMuxSrv() *muxSrv {
+	mux := &muxSrv{}
+
+	l, err := net.Listen("tcp", config.GetString("clientaddress"))
+	if err != nil {
+		log.WithError(err).Fatal("failed to create gd2-muxsrv listener")
+	}
+	mux.l = l
+	mux.m = cmux.New(l)
+
+	return mux
+}
+
+// Serve starts the handlers and the multiplexed listener
+func (m *muxSrv) Serve() {
+	log.Info("started muxsrv listener")
+	m.m.Serve()
```

Comment:
Both `m.m.Serve()` and `m.l.Close()` can return error. It'll be good to catch and  log these as well.